test/dynamo/test_misc.py

```diff
@@ -12854,9 +12855,11 @@ def test_torch_device_python_type(self):
             ("cpu", "cpu", None),
             ("cuda:0", "cuda", 0),
             ("hpu:0", "hpu", 0),
+            ("xpu:0", "xpu", 0),
         ]:
-            if (device == "cuda:0" and not TEST_CUDA) or (
-                device == "hpu:0" and not TEST_HPU
+            if (
+                device != "cpu"
+                and torch.accelerator.current_accelerator().type != device_type
```

Comment:
`torch.accelerator.current_accelerator` will return `None` if no accelerator detected. So in this situation, `.type` will crash.